app/Controllers/BaseController.php

```diff
@@ -24,7 +25,8 @@ class BaseController extends Controller
 	/**
 	 * Instance of the main Request object.
 	 *
-	 * @var IncomingRequest
+	 * @phpstan-ignore-next-line
+         * @var IncomingRequest&CLIRequest
```

Comment:
Ah my bad. Should this be `IncomingRequest|CLIRequest` instead? Because `&` means the property type can be `IncomingRequest` and `CLIRequest` at the same time.

Reply:
Just like in PHPUnit's mocking objects. It uses `&` between the `MockObject` and the mocked class to mean both types co-exist in the object mocked.

Reply:
that seems not detected in IDE? I tried with include RequestInteface and sems intelephense only detect RequestInterface, I use mobile interface so I can't check yet for only these 2 with pipe. could you verify, thank you.

Reply:
I'll check

Reply:
This one uses `IncomingRequest|CLIRequest` and testing on VSCode.

![image](https://user-images.githubusercontent.com/51850998/107855575-3a5bad80-6e5e-11eb-9299-07d465e81ae1.png)

Reply:
phpstan is no errors.

Reply:
thank you, I think that should go to `CodeIgniter\Controller` class instead.

Reply:
Yes, I think so.

Reply:
or not, as in `CodeIgniter\Controller` will got error, placing here seems fine.

Reply:
Just back to computer, trying in `CodeIgniter\Controller` got the following error:

```bash
  84     Property CodeIgniter\Controller::$request                            
         (CodeIgniter\HTTP\CLIRequest|CodeIgniter\HTTP\IncomingRequest) does  
         not accept CodeIgniter\HTTP\RequestInterface.
```

seems placing in this seems fine

Reply:
If you'll combine the three in pipes in `CodeIgniter\Controller` it will work `@var RequestInterface|IncomingRequest|CLIRequest`

Reply:
updated.

Reply:
@paulbalandan using `@var RequestInterface|IncomingRequest|CLIRequest` unfortunatelly not working on IDE ( I use VsCode):

![Screen Shot 2021-02-14 at 00 05 37](https://user-images.githubusercontent.com/459648/107856076-61af7c00-6e58-11eb-9860-9d86bc20f690.png)

Reply:
Ok. I see. I used `getPath()` and it turned red. Seems here is the right place.